benchmarks/entropy_adiabat/plugins/entropy_advection.cc

```diff
@@ -75,75 +79,94 @@ namespace aspect
-          const double rho_T              =
-            scratch.material_model_outputs.densities[q] *
-            scratch.material_model_inputs.temperature[q];
-
-          Assert (rho_T >= 0,
-                  ExcMessage ("The product of density and temperature needs to be a "
-                              "non-negative quantity."));
-
-          const double gamma =
-            scratch.heating_model_outputs.heating_source_terms[q];
-
-          const double field_term_for_rhs
-            = (use_bdf2_scheme ?
-               (scratch.old_field_values[q] *
-                (1 + time_step/old_time_step)
-                -
-                scratch.old_old_field_values[q] *
-                (time_step * time_step) /
-                (old_time_step * (time_step + old_time_step)))
-               :
-               scratch.old_field_values[q])
-              *
-              (rho_T);
-
-          Tensor<1,dim> current_u = scratch.current_velocity_values[q];
-          // Subtract off the mesh velocity for ALE corrections if necessary
-          if (this->get_parameters().mesh_deformation_enabled)
-            current_u -= scratch.mesh_velocity_values[q];
-
           const double JxW = scratch.finite_element_values.JxW(q);
 
-          // For the diffusion constant, use the larger of the physical
-          // and the artificial viscosity/conductivity/diffusion constant.
-          // One could also choose the sum of the two, but if the
-          // physical diffusion is larger than the artificial one,
-          // then (because the latter is chosen sufficiently large to
-          // make the problem stable) one may as well stick with the
-          // physical one. And if the physical diffusion is too small to
-          // make the problem stable, then we ought to choose the smallest
-          // diffusivity value that makes the problem stable -- which is
-          // exactly the artificial viscosity.
-          const double entropy_conductivity =  scratch.material_model_outputs.thermal_conductivities[q] * scratch.material_model_inputs.temperature[q] /
-                                               scratch.material_model_outputs.specific_heat[q];
-          const double diffusion_constant = std::max (entropy_conductivity,
-                                                      scratch.artificial_viscosity);
-
-          // do the actual assembly. note that we only need to loop over the advection
-          // shape functions because these are the only contributions we compute here
-          for (unsigned int i=0; i<advection_dofs_per_cell; ++i)
+          // solve the diffusion equation for the temperature
+          if (advection_field.is_temperature())
+            {
+              const double density_c_P = scratch.material_model_outputs.densities[q] *
+                                         scratch.material_model_outputs.specific_heat[q];
+
+              const double field_term_for_rhs = scratch.old_field_values[q] * density_c_P;
```

Comment:
Could we not use the same BDF2 time-stepping scheme that we use down in line 125? That would ensure the two separate solver schemes have the same convergence rate in time. It would probably not matter for practical results, but I would feel less concerned about the accuracy of the splitting we do.

Reply:
As we discussed, this does not work because we don't have anything we could use for the old_old_solution to make this second order.